src/lt/objs/deploy.cljs

```diff
@@ -21,31 +21,64 @@
 
 (def shell (load/node-module "shelljs"))
 (def fs (js/require "fs"))
+; Update the comment for the following to describe the full situation:
+(def original-fs (js/require "original-fs")) ;; Required per [fs - reading asar files fails · Issue #1658 · atom/electron](https://github.com/atom/electron/issues/1658)
 (def zlib (js/require "zlib"))
 (def request (load/node-module "request"))
-(def tar (load/node-module "tar"))
+(def tar (load/node-module "tar-fs"))
+(def zip (load/node-module "node-stream-zip"))
 (def home-path (files/lt-home ""))
 ;; TODO: get-proxy
 ;; (def get-proxy (.-App.getProxyForURL (js/require "nw.gui")))
 (def get-proxy)
 
-(defn tar-path [v]
+; TODO: Cleanup platform name code
+
+; TODO: Rename to release-file-extension
+(def tar-extension (if (platform/win?) ".zip" ".tar.gz"))
+
+(defn get-release-download-url
+  "Retrieve the download URL for the specified release version and this computer's platform."
+  [v cb]
+  (let [release-info-url (str "https://api.github.com/repos/LightTable/LightTable/releases/tags/"
+                              v)]
+    (fetch/xhr release-info-url
+               {}
+               (fn [data]
```

Comment:
@kenny-evitt The v is used for the rest / ajax call, I just didn't bother also using it in the callback traversing the result... as it would be very surprised if we had multiple versioned release artefacts for one release tag.

Both: Yeah I suppose @sbauer322 suggestion is a case where let would help readability.

I'll admit that I use threading quite extensively. Once you get used to them, they read actually quite a lot easier than the normal way to read Clojore/Lisp (inside->out). I think of them like a pipeline which you can just read sequentially.

Anyways I do think that using a let just to set up the following let that sets up the next let etc is a very imperative approach (and all those names doesn't really add that much, because they are just a means to an end - getting that download url. Come to think of it it could be extracted to a separate function) The multiple let approach just doesn't feel particularly functional to me. I guess that was my main point :-)